### PR TITLE
Clang format: add newlines for parameters and return types

### PR DESCRIPTION
**Pull request by robertlangenberg**
added two changes to the  .clang-format file:

AlwaysBreakAfterReturnType: All
which does what it says, the return type of a function is now always in a separate line. The function name in the next line is not indented anymore

and

BinPackArguments: 'false'
BinPackParameters: 'false'
AlignAfterOpenBracket: Align
which breaks function parameters in a separate line each if all parameters don't fit in a single line.

I would also have liked to add that multiple template arguments to a class or function also are in a separate line each, but that option doesn't seem to exist in clang-format. While I think it makes things easier to read, I'd rather not add it as a coding guideline as to not make contributing more complicated than necessary.

### CODE REVIEW

File: Core/include/Acts/EventData/ParameterConcept.hpp [hunk 1]
```diff
@@ -64,16 +64,19 @@ namespace concept {
...
-                surfacebound_momentum_exists, surfacebound_charge_exists,
-                surfacebound_time_exists, surfacebound_covariance_exists,
-                surfacebound_parameters_exists>;
+    constexpr static bool boundValue = require<reference_surface_exists,
+                                               surfacebound_position_exists,
+                                               surfacebound_momentum_exists,
+                                               surfacebound_charge_exists,
+                                               surfacebound_time_exists,
+                                               surfacebound_covariance_exists,
+                                               surfacebound_parameters_exists>;
```
Comment (paulgessinger) [inline Core/include/Acts/EventData/ParameterConcept.hpp:79]: We used to have this, right? I'm personally not a huge caon of this.

File: Core/include/Acts/EventData/ParameterSet.hpp [hunk 2]
```diff
@@ -95,7 +95,9 @@ class ParameterSet {
       detail::are_sorted<true, true, parameter_indices_t, params...>::value,
       "parameter identifiers are not sorted");
   static_assert(
-      detail::are_within<unsigned int, 0, kSizeMax,
+      detail::are_within<unsigned int,
+                         0,
+                         kSizeMax,
```
Comment (paulgessinger) [inline Core/include/Acts/EventData/ParameterSet.hpp:100]: Same here, I find this appalling 😄

File: Core/include/Acts/EventData/ParameterConcept.hpp (hunk 1 above)
Comment (robertlangenberg) [reply Core/include/Acts/EventData/ParameterConcept.hpp:79]: I'm surprised you took an example where it's so obviously much more readable :D
If people really disagree with how this block looks before/after then there's nothing I can add to that, so I'm happy to use this as advertisement. Please cast your vote as reaction to the PR description with a thumbs up or down. voting closes next tuesday at 16:59 :)

Comment (robertlangenberg) [reply Core/include/Acts/EventData/ParameterConcept.hpp:79]: If you'd like to change something else or modify the changed bits, please write them as a comment and we'll cast our vote on that until next tuesday as well.

**Pull request closed without merging**